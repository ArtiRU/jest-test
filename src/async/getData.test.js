const {it, expect, describe, beforeEach, mockReturnValue, test, afterEach} = require("@jest/globals");
const getData = require("./getData")
const axios = require("axios");
const getMockUsers = require("./__mocks__/users");



jest.mock("axios")

describe("Testing getData", () => {
    let response;

    beforeEach(() => {
        response = {
            data: getMockUsers()
        }
    })

    test("getData to equal [1, 2, 3]", async () => {
        axios.get.mockReturnValue(response)

        const data = await getData();

        expect(data).toEqual([1, 2, 3])
    })

    test("getData snapshot", async () => {
        axios.get.mockReturnValue(response)

        const data = await getData()

        expect(axios.get).toBeCalledTimes(3)
        expect(data).toMatchSnapshot();
    })

    afterEach(() => {

    })
})