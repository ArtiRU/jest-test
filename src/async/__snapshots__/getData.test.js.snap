// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing getData getData snapshot 1`] = `
[
  1,
  2,
  3,
]
`;
